src/controllers/user.js

```diff
@@ -1,20 +1,53 @@
+const UserModel = require('../models/user');
+
 module.exports = {
-  getAllUsers: (req, res) => {
-    res.send('All users');
+  getAllUsers: async (req, res) => {
+    try {
+      const users = await UserModel.find();
+      res.json(users);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
   },
-  getSingleUser: (req, res) => {
-    res.send('Single user');
+  getSingleUser: async (req, res) => {
+    const { id } = req.params;
+    try {
+      const user = await UserModel.findById(id);
+      if (!user) throw new Error("The user with the specified ID wasn't found");
+      res.json(user);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
```

Comment:
Was the main branch merged correctly? Because I think this function was updated in another PR that has been merged to main.

Reply:
User endpoints are not merged yet. I will delete getSingleUser, but I have to leave getAllUsers and updateUser for my tests to pass.